lib/pure/ioselects/ioselectors_select.nim

```diff
@@ -438,16 +439,17 @@ template withData*[T](s: Selector[T], fd: SocketHandle|int, value,
                       body1, body2: untyped) =
   mixin withSelectLock
   s.withSelectLock():
-    var value: ptr T
-    let fdi = int(fd)
-    var i = 0
-    while i < FD_SETSIZE:
-      if s.fds[i].ident == fdi:
-        value = addr(s.fds[i].data)
-        break
-      inc(i)
-    if i != FD_SETSIZE:
-      body1
-    else:
-      body2
+    block:
```

Comment:
Why was this 'block' introduced? Remove it.

Reply:
The block was added because the `processBasicCallbacks` template in async dispatch https://github.com/ehmry/Nim/blob/cc3888de30d7b7609a535b4be93c23c7aeaf1d9b/lib/pure/asyncdispatch.nim#L1136 calls `withData` twice with the same symbol names, this expands to the same symbol being defined twice in the same scope, unless `withData` opens a new scope with `block`.

Reply:
Why is this only an issue for Genode?

Reply:
Windows too, but I've never tried Nim on windows https://github.com/ehmry/Nim/blob/cc3888de30d7b7609a535b4be93c23c7aeaf1d9b/lib/pure/selectors.nim#L302

Reply:
The issue only occurs for Genode because it is the only platform that uses `select` and the non-windowds half of `asyncdispatch`.